src/buildstream/_frontend/cli.py

```diff
@@ -629,19 +654,44 @@ def show(app, elements, deps, except_, order, format_):
     \b
         bst show target.bst --format \\
             $'---------- %{name} ----------\\n%{vars}'
+
+    **JSON OUTPUT**
+
+    The ``--json`` flag will cause bst to output information in machine readable
+    JSON. When using this flag you may also specify ``--kind`` to control the
+    type of information that is output.
+
+    To dump everything:
+    \b
+        bst show --json --kind all
     """
     with app.initialized():
 
+        if as_json and format_:
+            raise AppError("--format and --json are mutually exclusive")
+
+        if format_ and kind:
+            raise AppError("--format does not support --kind")
```

Comment:
Not sure I like this error message, as I see `--format` and `--kind` (in your current implementation) to be essentially the same thing except `--format` is used in the default mode and `--kind` is used in the json mode.

Maybe we could change it such that `--format` could take the special value `json`, and that would unlock the (replacement for the) `--kind` option.

Let's wait for other opinions.